crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,234 @@
+function CrowdsourceHinter(runtime, element, data){
+    //switching units back to a previous question will make a second hinter start up.
+    //executeHinter is used to disable the hinter after switching units in an edX course.
+    var executeHinter = true;
```

Comment:
I'd be interested in cleaning this up. This should be stateless. We should inspect what is happening, and only do steps if required, rather than holding initial state.